auth/forward.go

```diff
@@ -0,0 +1,47 @@
+package auth
+
+import (
+	"io/ioutil"
+	"net/http"
+
+	"github.com/containous/traefik/log"
+	"github.com/containous/traefik/types"
+)
+
+// Forward the authentication to a external server
+func Forward(forward *types.Forward, w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+	client := http.Client{}
+
+	forwardReq, err := http.NewRequest("GET", forward.Address, nil)
+	if err != nil {
+		log.Debugf("Error calling %s. Cause %s", forward.Address, err)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+	forwardReq.Header = r.Header
+
+	forwardResponse, forwardErr := client.Do(forwardReq)
+	if forwardErr != nil {
+		log.Debugf("Error calling %s. Cause: %s", forward.Address, forwardErr)
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	defer forwardResponse.Body.Close()
```

Comment:
I would move this line after the block:
```
if readError != nil {
...
}
defer forwardResponse.Body.Close()
```
Indeed, if `readError` is not nil, maybe the body will be empty, then the defered `Close()` would lead to a panic.